src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.CompilationTrackerState.cs

```diff
@@ -90,13 +45,33 @@ public WithCompilationTrackerState(Compilation compilationWithoutGeneratedDocume
                     }
 #endif
                 }
+
+                /// <summary>
+                /// Returns a <see cref="AllSyntaxTreesParsedState"/> if <paramref name="intermediateProjects"/> is
+                /// empty, otherwise a <see cref="InProgressState"/>.
+                /// </summary>
+                public static CompilationTrackerState Create(
+                    Compilation compilationWithoutGeneratedDocuments,
+                    CompilationTrackerGeneratorInfo generatorInfo,
+                    Compilation? compilationWithGeneratedDocuments,
+                    ImmutableList<(ProjectState state, CompilationAndGeneratorDriverTranslationAction action)> intermediateProjects)
+                {
+                    Contract.ThrowIfTrue(intermediateProjects is null);
+
+                    // If we don't have any intermediate projects to process, just initialize our
+                    // DeclarationState now. We'll pass false for generatedDocumentsAreFinal because this is being called
+                    // if our referenced projects are changing, so we'll have to rerun to consume changes.
```

Comment:
Not related to this PR, but this comment has not aged well with refactorings. (DeclarationState we renamed ages ago...)